CMakeLists.txt

```diff
@@ -5,6 +5,14 @@ if ("${CMAKE_SOURCE_DIR}" STREQUAL "${CMAKE_BINARY_DIR}")
   message(FATAL_ERROR "In-source builds are prohibited.")
 endif ("${CMAKE_SOURCE_DIR}" STREQUAL "${CMAKE_BINARY_DIR}")
 
+# Import MFEM
+find_package(mfem REQUIRED NAMES MFEM)
+# Use the same C++ compiler as MFEM. This is needed when MFEM was built using
+# an MPI wrapper and we do not have explicitly the MPI compile and link flags.
+if (NOT CMAKE_CXX_COMPILER AND MFEM_CXX_COMPILER)
+  set(CMAKE_CXX_COMPILER "${MFEM_CXX_COMPILER}")
```

Comment:
I don't think this will actually do anything. See, for example, [this](https://cmake.org/Wiki/CMake_FAQ#How_do_I_use_a_different_compiler.3F) or [this](https://cmake.org/cmake/help/v3.7/variable/CMAKE_LANG_COMPILER.html). Because the call occurs after the call to `project()`, it should have no effect. If you really, really, really are set on doing it this way, you need to force the change in the CMake cache, by doing something like this:

    set(CMAKE_CXX_COMPILER "${MFEM_CXX_COMPILER}" CACHE FILEPATH "A really bad idea." FORCE)

and that should actually make the change happen in the CMakeCache. Just calling bland ol' `set()` won't actually change the value that gets written to the CMakeCache.txt file.

Moreover, I know, I know, setting `CMAKE_CXX_COMPILER=mpicxx` is "easy" and "convenient," but it just isn't the right thing to do with CMake, see [this](https://cmake.org/pipermail/cmake/2011-June/045037.html) or [this](https://stackoverflow.com/questions/23163075/how-to-compile-an-mpi-included-c-program-using-cmake). I understand that there are some rather special MPI "compilers" out there that CMake doesn't know about, but these are certainly exceptional cases. Users should be encouraged to *not* set the `CMAKE_<LANG>_COMPILER` variables at all and use `MPI_<LANG>_COMPILER` and/or `MPI_<LANG>_INCLUDE_PATH` and `MPI_<LANG>_LIBRARIES` if needed as described in the [CMake documentation](https://cmake.org/cmake/help/v3.7/module/FindMPI.html#usage).